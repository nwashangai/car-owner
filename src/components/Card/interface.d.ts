export interface CardProps {
  classes?: string;
  header?: string;
  children: any;
  onClick?: any;
}
