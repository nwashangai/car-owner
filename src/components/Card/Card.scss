@import 'sass-rem';
@import 'assets/variables';

.card-theme {
  background: map-get($palette, pure);
  border: 1px solid map-get($palette, border-line);
  box-shadow: 0 0 rem(10px) 0 map-get($palette, border-shadow);
}
