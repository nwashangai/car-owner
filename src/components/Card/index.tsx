// react libraries
import * as React from 'react';

// Interface
import { CardProps } from './interface';

// Styles
import './Card.scss';

export default (props: CardProps) => {
  const { classes, children, ...rest } = props;
  return (
    <div className={`card-theme ${classes}`} {...rest}>
      {children}
    </div>
  );
};
