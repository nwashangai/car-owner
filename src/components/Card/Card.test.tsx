// react libraries
import * as React from 'react';

// third-party libraries
import { mount } from 'enzyme';

// components
import Card from './';

describe('The Card component', () => {
  const props = {
    header: 'header',
    classes: 'my-class'
  };

  it('displays card with child props', () => {
    const wrapper = mount(
      <Card {...props}>
        <div className="my-child">child component</div>
      </Card>
    );

    const child = wrapper.find('.my-child');
    expect(child).toMatchSnapshot();
    expect(child.props().className).toBe('my-child');
  });
});
