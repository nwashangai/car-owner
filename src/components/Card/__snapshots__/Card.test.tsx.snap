// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Card component displays card with child props 1`] = `ReactWrapper {}`;
